PerformanceTests/JetStream2/JetStreamDriver.js

```diff
@@ -640,75 +660,158 @@ class Benchmark {
+                    this.preloads.push([ blobData.prop, blobData.blobURL ]);
+                    this.updateCounter();
+                }).catch((error) => {
+                    // We'll try again later in retryPrefetchResourceForBrowser(). Don't throw an error.
+                    if (!this.failedPreloads)
+                        this.failedPreloads = { };
+                    this.failedPreloads[prop] = true;
+                    JetStream.counter.failedPreloadResources++;
+                }));
+            }
+        }
 
-                ++counter.loadedFiles;
-                var statusElement = document.getElementById("status");
-                statusElement.innerHTML = `Loading ${counter.loadedFiles} of ${counter.totalFiles} ...`;
-            });
-        }));
+        JetStream.counter.totalResources += promises.length;
+        return Promise.all(promises);
     }
 
-    fetchResources() {
-        if (this._resourcesPromise)
-            return this._resourcesPromise;
+    async retryPrefetchResource(type, prop, file) {
+        let counter = JetStream.counter;
+        var blobData = JetStream.blobDataCache[file];
+        if (blobData.blob) {
+            // The same preload blob may be used by multiple subtests. Though the blob is already loaded,
+            // we still need to check if this subtest failed to load it before. If so, handle accordingly.
+            if (type == "preload") {
+                if (this.failedPreloads && this.failedPreloads[blobData.prop]) {
+                    this.failedPreloads[blobData.prop] = false;
+                    this.preloads.push([ blobData.prop, blobData.blobURL ]);
+                    counter.failedPreloadResources--;
+                }
+            }
+            return !counter.failedPreloadResources && counter.loadedResources == counter.totalResources;
+        }
 
-        let filePromises = !isInBrowser ? this.plan.files.map((file) => fileLoader.load(file)) : [];
-        let preloads = [];
-        let preloadVariableNames = [];
+        // Retry fetching the resource.
+        JetStream.loadCache[file] = null;
+        await this.loadBlob(type, prop, file, false).then((blobData) => {
+            if (!window.allIsGood)
+                return;
+            if (blobData.type == "preload")
+                this.preloads.push([ blobData.prop, blobData.blobURL ]);
+            this.updateCounter();
+        });
+
+        if (!blobData.blob) {
+            window.allIsGood = false;
+            throw new Error("Fetch failed"); 
+        }
+
+        return !counter.failedPreloadResources && counter.loadedResources == counter.totalResources;
+    }
+
+    async retryPrefetchResourcesForBrowser() {
+        if (!isInBrowser)
+            return;
 
-        if (isInBrowser && this.plan.preload) {
+        let counter = JetStream.counter;
+        for (let resource of this.plan.files) {
+            let allDone = await this.retryPrefetchResource("file", null, resource);
```

Comment:
Are we sure that we don't want to `Promise.all(this.plan.files.map())` here?

Reply:
Yes, we’re deliberately downgrading to sequential 1 by 1 fetch so to go easy on the network.  We already tried fetching everything all at once in `prefetchResourcesForBrowser` and fail.  That’s why we’re in this function.